File: java_brain/src/main/java/org/netbrains/javabrain/activation/Sigmoid.java
package org.netbrains.javabrain.activation;

public class Sigmoid extends Activation {
    @Override
    public double activate(double x) {
        return 1.0 / (1 + Math.pow(Math.E, -x));
    }

    @Override
    public double derivative(double x) {
        double val = activate(x);
        return val * (1 - val);
    }
}

File: swift_brains/swift_brains/Sigmoid.swift

import Darwin

class Sigmoid : Activation {
    private let e = 2.718
    
    public override func activate(x: Double) -> Double {
        return 1.0 / (1 + pow(e, -x))
    }
    
    public override func derivative(x: Double) -> Double {
        let a = activate(x: x)
        return a * (1 - a)
    }
}
